orchestra/model/component.py

```diff
@@ -1,38 +1,47 @@
-from typing import Dict
+import hashlib
+from typing import Dict, Union
 
-from . import build
-from ..actions import CloneAction
+from . import build as bld
+from ..actions import any_of
+from ..actions import clone
 
 
 class Component:
-    def __init__(self,
-                 name: str,
-                 license: str,
-                 from_source: bool,
-                 binary_archives: str,
-                 skip_post_install=False,
-                 ):
+    def __init__(
+            self,
+            name: str,
+            serialized_component,
+            configuration
+    ):
         self.name = name
-        self.builds: Dict[str, 'build.Build'] = {}
-        self.skip_post_install = skip_post_install
-        self.license = license
-        self.from_source = from_source
-        self.clone: CloneAction = None
-        self.binary_archives = binary_archives
-        self.self_hash = None
-        self.recursive_hash = None
-        self._default_build = None
-
-    def add_build(self, bld: 'build.Build', default=False):
-        self.builds[bld.name] = bld
-        if default:
-            self._default_build = bld
+        self.builds: Dict[str, bld.Build] = {}
+        self.skip_post_install = serialized_component.get("skip_post_install", False)
+        self.license = serialized_component.get("license")
+        self.binary_archives = serialized_component.get("binary_archives")
+        self.build_from_source = serialized_component.get("build_from_source", False)
+        self.add_to_path = serialized_component.get("add_to_path", [])
+        self.repository = serialized_component.get("repository")
```

Comment:
What if we do this more like:

```python
lel(serialized_component,
     [("skip_post_install", False),
     ("license", ...),
     ...])
```

And `lel` does the right thing.
Less error prone.

Reply:
I did try various things and this was one of them, I did not go for it being afraid of writing code that is too smart and uses getprop. Making the change :+1:

Reply:
The issue with that approach is that it requires using `setprop` with a level of indirection, and the IDEs/static analyzers will not recognize the properties anymore.

Reply:
Sad but OK